docs/repo/property-pages/property-value-interception.md

```diff
@@ -0,0 +1,73 @@
+﻿# Property Value Interception
+
+Sometimes it is necessary to intercept reads and writes of property values, and modify those values while they are in flight. Property value interception allows this.
+
+## Mechanism
+
+The property must opt in to interception. Its `DataSource` must specify the `Persistence` property as one of:
+
+- `ProjectFileWithInterception` writes values to the project file.
+- `UserFileWithInterception` writes values to the project's `.user` file.
+
+For example:
+
+```xml
+<StringProperty Name="MyProperty"
+                DisplayName="My property"
+                ...>
+  <StringProperty.DataSource>
+    <DataSource PersistedName="MyProperty"
+                Persistence="ProjectFileWithInterception"
+                HasConfigurationCondition="False"
+                ... />
+  </StringProperty.DataSource>
+  ...
+</StringProperty>
+```
+
+> See [Property Specification](property-specification.md) for more information on specifying properties.
+
+There must be a corresponding export of `IInterceptingPropertyValueProvider` for the property. For example:
+
+```c#
+[ExportInterceptingPropertyValueProvider("MyProperty", ExportInterceptingPropertyValueProviderFile.ProjectFile)]
```

Comment:
Maybe also mention here how multiple attributes could be used to intercept several properties or that a string array could be passed as a parameter.

Reply:
Thank you, that's definitely something we should cover. I think we could make more use of this feature in this repo, actually.